src/main/java/com/artipie/jfr/JfrSlice.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2020-2021 artipie.com
+ * https://github.com/artipie/artipie/LICENSE.txt
+ */
+package com.artipie.jfr;
+
+import com.artipie.http.Connection;
+import com.artipie.http.Headers;
+import com.artipie.http.Response;
+import com.artipie.http.Slice;
+import com.artipie.http.rq.RequestLineFrom;
+import com.artipie.http.rs.RsStatus;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import java.util.concurrent.CompletionStage;
+import java.util.function.BiConsumer;
+import org.reactivestreams.Publisher;
+
+/**
+ * Slice wrapper to generate JFR events for every the {@code response} method call.
+ *
+ * @since 0.28.0
+ * @checkstyle LocalFinalVariableNameCheck (500 lines)
+ */
+public final class JfrSlice implements Slice {
+
+    /**
+     * Original slice.
+     */
+    private final Slice original;
+
+    /**
+     * Ctor.
+     *
+     * @param original Original slice.
+     */
+    public JfrSlice(final Slice original) {
+        this.original = original;
+    }
+
+    @Override
+    public Response response(
+        final String line,
+        final Iterable<Map.Entry<String, String>> head,
+        final Publisher<ByteBuffer> body
+    ) {
+        final RequestLineFrom rqLine = new RequestLineFrom(line);
+        final SliceResponseEvent event = new SliceResponseEvent();
+        event.method = rqLine.method().value();
+        event.path = rqLine.uri().getPath();
+        event.begin();
+        final Response res = this.original.response(
+            line,
+            head,
+            new ChunksAndSizeMetricsPublisher(
+                body,
+                (chunks, size) -> {
+                    event.requestChunks = chunks;
+                    event.requestSize = size;
+                }
+            )
+        );
+        return new JfrResponse(
+            res,
+            (chunks, size) -> {
+                event.responseChunks = chunks;
+                event.responseSize = size;
+                event.commit();
```

Comment:
Please correct me if I see imposible case or not may be it is not important case, any way I describe my point:
Lets image that during some error happens and the callback function on line 65 will not be called.

In this case we will have input traffic not mentioned in Jfr statistics.
We can consider the definition of two JFR events: SliceRequestEvent and SliceResponseEvent:
 1. SliceRequestEvent {
    public volatile String method;
    public volatile String path;
    public volatile int requestChunks;
    public volatile long requestSize;
 }
 2. SliceResponseEvent {
    public volatile String method;
    public volatile String path;
    public volatile int responseChunks;
    public volatile long responseSize;
 }
 In case error in Jfr statistics will be present SliceRequestEvent.
 
PS: I mention that it is just theoretic thoughts that may be not important for this case.